apic_ml2/neutron/services/l3_router/l3_apic.py

```diff
@@ -13,6 +13,7 @@
 #    License for the specific language governing permissions and limitations
```

Comment:
This is already done actually and also being tested inside UT test_router_lifecycle(). By declaring this:
use_routing_context = sa.Column(sa.String(36), sa.ForeignKey('routers.id', ondelete="RESTRICT"))
The first router can't be deleted when there is a 2nd router pointing to the 1st router thru the use_routing_context field.